src/ripple/rpc/impl/TransactionSign.cpp

```diff
@@ -559,17 +559,9 @@ transactionConstructImpl(
     std::pair<Json::Value, Transaction::pointer> ret;
 
     // Turn the passed in STTx into a Transaction.
-    Transaction::pointer tpTrans;
-    {
-        std::string reason;
-        tpTrans = std::make_shared<Transaction>(stpTrans, reason, app);
-        if (tpTrans->getStatus() != NEW)
-        {
-            ret.first = RPC::make_error(
-                rpcINTERNAL, "Unable to construct transaction: " + reason);
-            return ret;
-        }
-    }
+    auto tpTrans = std::make_shared<Transaction>(stpTrans);
+    assert(tpTrans);
```

Comment:
As above, if `std::make_shared` fails, it'll throw, so if the execution reaches the `assert`, then `tpTrans` will be set. I think it makes more sense to `assert(tpTrans->getStatus() == NEW);`, or since `NEW` is the default ctor parameter, remove the assert entirely.